src/it/scala/net/snowflake/spark/snowflake/IntegrationEnv.scala

```diff
@@ -240,17 +240,19 @@ trait IntegrationEnv
     jdbcUpdate("alter session set timezone='UTC'")
 
     // Use fewer partitions to make tests faster
-    sparkSession = SparkSession.builder
-      .master("local")
-      .appName("SnowflakeSourceSuite")
-      .config("spark.sql.shuffle.partitions", "6")
-      // "spark.sql.legacy.timeParserPolicy = LEGACY" is added to allow
-      // spark 3.0 to support legacy conversion for unix_timestamp().
-      // It may not be necessary for spark 2.X.
-      .config("spark.sql.legacy.timeParserPolicy", "LEGACY")
-      .getOrCreate()
+    sparkSession = createDefaultSparkSession
   }
 
+  protected def createDefaultSparkSession = SparkSession.builder
```

Comment:
why move it out?
if necessary can we use `lazy protected val ...` to replace `protected var sparkSession: SparkSession = _`?

Reply:
In the unit test, I need to create a SparkSession with extra Spark options.
To create a SparkSession, I need to stop the current SparkSession first.
Extract this as a separate function so that I can create the default SparkSession after the test case done.